.github/scripts/mr_publish_results.py

```diff
@@ -0,0 +1,64 @@
+import analytics
+import datetime
+import json
+import os
+
+analytics.write_key = os.environ['SEGMENT_TOKEN']
+
+def send_to_segment(context):
+    jobID = os.environ['GITHUB_RUN_ID']
+
+    analytics.identify(jobID, {
+        'name': 'model-regression-tests',
+        'created_at': datetime.datetime.now()
+    })
+
+    analytics.track(jobID, 'results', {
+        'dataset': os.environ['DATASET_NAME'],
+        'workflow': os.environ['GITHUB_WORKFLOW'],
+        'config': os.environ['CONFIG'],
+        'runner_type': os.environ['RUNNER_TYPE'],
+        'test_run_time': os.environ['TEST_RUN_TIME'],
+        'train_run_time': os.environ['TRAIN_RUN_TIME'],
+        'total_run_time': os.environ['TOTAL_RUN_TIME'],
+        'event': os.environ['GITHUB_EVENT_NAME'],
+        **context
+    })
+
+def read_results(file):
+    result = {}
+    with open(file) as json_file:
+        data = json.load(json_file)
+        
+        if "accuracy" in data:
+            result["accuracy"] = data["accuracy"]
+
+        if "weighted avg" in data:
+            result["weighted_avg"] = data["weighted avg"]
+
+        if "macro avg" in data:
+            result["macro_avg"] = data["macro avg"]
+
+        if "micro avg" in data:
+            result["micro_avg"] = data["micro avg"]
```

Comment:
let's do something like this:

```suggestion
    with open(file) as json_file:
        data = json.load(json_file)
        
        keys = ["accuracy", "weighted avg", macro avg", "micro avg"]
        result = {key: data[key] for key in keys if key in data}
```